insights/tests/test_specs_save_as.py

```diff
@@ -0,0 +1,210 @@
+    smpl_file_w_filter = RegistryPoint(filterable=True)
+    first_of_spec_w_filter = RegistryPoint(filterable=True)
+
+
+class Stuff(Specs):
+    many_glob = glob_file(here + "/*.py", save_as=SAVE_AS_MAP['many_glob'][0])
+
+    @datasource(HostContext)
+    def files(broker):
+        """ Return a list of directories from the spec filter """
+        return [this_file, "/etc/os-release"]
+
+    many_foreach = foreach_collect(files, "%s", save_as=SAVE_AS_MAP['many_foreach'][0])
+    smpl_cmd = simple_command("/usr/bin/uptime", save_as=SAVE_AS_MAP['smpl_cmd'][0])
+    smpl_cmd_w_filter = simple_command("echo -n ' hello '", save_as=SAVE_AS_MAP['smpl_cmd_w_filter'][0])
+    smpl_file = simple_file(this_file, save_as=SAVE_AS_MAP['smpl_file'][0])
+    smpl_file_w_filter = simple_file(this_file, kind=RawFileProvider, save_as=SAVE_AS_MAP['smpl_file_w_filter'][0])
+    first_of_spec_w_filter = first_file([this_file, "/etc/os-release"], save_as=SAVE_AS_MAP['first_of_spec_w_filter'][0])
+
+
+class stage(dr.ComponentType):
+    def invoke(self, broker):
+        return self.component(broker)
+
+#
+# TEST
+#
+
+
+def teardown_function(func):
+    filters._CACHE = {}
+
+    if func in [test_specs_save_as_collect, test_specs_save_as_no_collect]:
+        del filters.FILTERS[Stuff.smpl_cmd_w_filter]
+        del filters.FILTERS[Stuff.smpl_file_w_filter]
+        del filters.FILTERS[Stuff.first_of_spec_w_filter]
+
+
+@stage(Stuff.many_glob,
+       Stuff.many_foreach,
+       Stuff.smpl_file,
+       Stuff.smpl_file_w_filter,
+       Stuff.smpl_cmd,
+       Stuff.smpl_cmd_w_filter,
+       Stuff.first_of_spec_w_filter)
+def dostuff(broker):
+    assert Stuff.many_glob in broker
+    assert Stuff.many_foreach in broker
+    assert Stuff.smpl_cmd in broker
+    assert Stuff.smpl_cmd_w_filter in broker
+    assert Stuff.smpl_file in broker
+    assert Stuff.smpl_file_w_filter in broker
+    assert Stuff.first_of_spec_w_filter in broker
+
+
+def test_specs_save_as_no_collect():
+    add_filter(Stuff.smpl_cmd_w_filter, " hello ")
+    add_filter(Stuff.smpl_file_w_filter, "def test")
+    add_filter(Stuff.first_of_spec_w_filter, [" hello ", "def test"])
+    broker = dr.Broker()
+    broker[HostContext] = HostContext()
+    broker = dr.run(dr.get_dependency_graph(dostuff), broker)
+    # general tests
+    assert dostuff in broker, broker.tracebacks
+    assert broker[Stuff.smpl_file].content == file_content
+    assert not any(l.endswith("\n") for l in broker[Stuff.smpl_file].content)
+    assert "hello" in broker[Stuff.smpl_cmd_w_filter].content[0]
+    assert len(broker[Stuff.smpl_cmd_w_filter].content) == 1  # RawFileProvdier
+    assert len(broker[Stuff.first_of_spec_w_filter].content) == 9
+    # test "Save As"
+    assert broker[Stuff.many_glob][-1].save_as == SAVE_AS_MAP['many_glob'][1]
+    assert broker[Stuff.many_foreach][0].save_as == SAVE_AS_MAP['many_foreach'][1]
+    assert broker[Stuff.many_foreach][1].save_as == SAVE_AS_MAP['many_foreach'][1]
```

Comment:
Run `pytest` in my local (MacOS), and met this following error: 
```
insights/tests/test_specs_save_as.py::test_specs_save_as_no_collect FAILED
insights/tests/test_specs_save_as.py::test_specs_save_as_collect[True] PASSED
insights/tests/test_specs_save_as.py::test_specs_save_as_collect[False] PASSED

============================================================ FAILURES ============================================================
_________________________________________________ test_specs_save_as_no_collect __________________________________________________

    def test_specs_save_as_no_collect():
        ...
        # test "Save As"
        assert broker[Stuff.many_glob][-1].save_as == SAVE_AS_MAP['many_glob'][1]
        assert broker[Stuff.many_foreach][0].save_as == SAVE_AS_MAP['many_foreach'][1]
>       assert broker[Stuff.many_foreach][1].save_as == SAVE_AS_MAP['many_foreach'][1]
E       IndexError: list index out of range

insights/tests/test_specs_save_as.py:154: IndexError
```
Just in case `/etc/os-release` no exist in some hosts, what about do a double check before the assert of this file, eg: 
```suggestion
    if len(broker[Stuff.many_foreach]) > 1:
        assert broker[Stuff.many_foreach][1].save_as == SAVE_AS_MAP['many_foreach'][1]
```